Source Code/Visual C++ Project/Utilities.c

```diff
@@ -118,17 +119,21 @@ void PA_SetTextParameter(PA_PluginParameters params, short index, char* text, LO
 {
 	PA_Unistring		*UnistringParam;
 	PA_Unichar			*translatedText;
-	LONG_PTR			length;
+	LONG_PTR			length; 
+	size_t              convertedChars; // ZRW 11/17/16 WIN-39
+	wchar_t             junk;  // ZRW 12/1/16 WIN-39
 
 	if (len > 0){
-		length = mbstowcs(0, text, len) + 1;
-
-		translatedText = malloc(length * sizeof(PA_Unichar));
+		// ZRW 11/17/16 WIN-39 mbstowcs -> mbstowcs_s
+		mbstowcs_s(&convertedChars, &junk, 255, text, len);
```

Comment:
You do not need to call mbstowcs_s twice, simply allocate (len + 1) as only 1 byte will be needed per ASCII character.